mne/channels/channels.py

```diff
@@ -740,7 +740,23 @@ def pick_types(self, meg=None, eeg=False, stim=False, eog=False,
             ias=ias, syst=syst, seeg=seeg, dipole=dipole, gof=gof, bio=bio,
             ecog=ecog, fnirs=fnirs, include=include, exclude=exclude,
             selection=selection)
-        return self._pick_drop_channels(idx)
+
+        res = self._pick_drop_channels(idx)
```

Comment:
I **think** you should be able to just do:

```suggestion
       self._pick_drop_channels(idx)
```
(it works in-place)

and `return self` at the end of the method call